src/main/java/nl/novi/techiteasy/services/TelevisionService.java

```diff
@@ -0,0 +1,89 @@
+package nl.novi.techiteasy.services;
+
+import nl.novi.techiteasy.dtos.TelevisionOutPutDto;
+import nl.novi.techiteasy.dtos.TelevisionInputDto;
+import nl.novi.techiteasy.exceptions.RecordNotFoundException;
+import nl.novi.techiteasy.models.Television;
+import nl.novi.techiteasy.repository.TelevisionRepository;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class TelevisionService {
+
+    private final TelevisionRepository repo;
+
+    public TelevisionService(TelevisionRepository repos){this.repo = repos;}
+
+
+    public TelevisionInputDto createTelevision(TelevisionInputDto televisionInputDto) {
+     repo.save(fromDtoToTelevision(televisionInputDto));
+        return televisionInputDto;
+    }
+
+    public List<TelevisionOutPutDto> GetAllTelevision() {
+        List<Television> televisionList = repo.findAll();
+        List<TelevisionOutPutDto> televisionOutPutDtoList = new ArrayList<>();
+
+        for(Television television : televisionList) {
+            televisionOutPutDtoList.add(fromTelevisionToDto(television));
+        }
+        return televisionOutPutDtoList;
+    }
+
+    public TelevisionOutPutDto getTelevisionById(Long id) {
+
+       Optional<Television> tv = repo.findById(id);
+
+        if(tv.isPresent()){
+            return fromTelevisionToDto(tv.get());
+        }
+        else {
+            throw new RecordNotFoundException("no tv founded");
+        }
+    }
+
+    public TelevisionOutPutDto updateTelevision(Long id, TelevisionInputDto televisionInputDto) {
+        Optional<Television> optionalTelevision = repo.findById(id);
+
+        if (optionalTelevision.isPresent()) {
+            Television existingTelevision = optionalTelevision.get();
+            existingTelevision.setName(televisionInputDto.name);
+            existingTelevision.setPrice(televisionInputDto.price);
+
+            repo.save(existingTelevision);
+
+            return fromTelevisionToDto(existingTelevision);
+        } else {
+            throw new RecordNotFoundException("Television with id " + id + " not found");
+        }
+    }
+    public void deleteTelevision(Long id) {
+        Optional<Television> optionalTelevision = repo.findById(id);
+
+        if (optionalTelevision.isPresent()) {
+            repo.deleteById(id);
+        } else {
+            throw new RecordNotFoundException("Television with id " + id + " not found");
+        }
+    }
+
+    public  TelevisionOutPutDto fromTelevisionToDto(Television television){
+        TelevisionOutPutDto dto = new TelevisionOutPutDto();
+        dto.id = television.getId();
+        dto.name = television.getName();
+        dto.price = television.getPrice();
+        return dto;
+    }
+
+    public Television fromDtoToTelevision(TelevisionInputDto dto){
```

Comment:
Als je meer de verantwoordelijkheden wil scheiden dan zou ik het mappen naar een aparte mapper verplaatsen. De service is voor het toepassen van business logica.

Reply:
is het beter om een static method van te maken en in de TelevisionInputDto zelf te zetten @nairu777